### PR TITLE
First attempt at centering the grid and inserting rows/column labels

### LINKED ISSUE #1: Centering
Grid and text above it should be centered.

### PR DESCRIPTION
**Pull request by kerznal**
This attempts to resolve #1 and #3. I suspect this isn't the best approach to centering (having a dummy block, and constantly having to use the magic number "200px" seem not robust). I also suspect that constantly using the same props in Typography components is not ideal.

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -64,34 +64,50 @@ function App() {
...
-        </Box>
-      </Container>
+        <Container sx={{ py: 2, flex: 1}}>
+          <Typography variant="h4" sx = {{textAlign: 'center'}}>Hello</Typography>
+          <Typography variant="body1" sx = {{textAlign: 'center'}}>Explanation stuff</Typography>
+          <Stack alignItems='center'>
+            <Stack direction='row' alignItems={'center'}>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>A category</Typography>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>Some other category</Typography>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>Looooooooong looooooong multi-line category</Typography>
```
Comment (mattrafuse) [inline ui/src/App.tsx]: I'd use maxWidth instead. Also consider percentages or `vw` metrics. Alternatively, simply set the padding/margin to whatever feels comfortable and let the browser figure out the rest. Not awful tbh

File: ui/src/App.tsx [hunk 2]
```diff
@@ -64,34 +64,50 @@ function App() {
...
-          })}
-        >
-          {[...Array(9).keys()].map(entry => <GridButton key={entry}/>)}
-        </Box>
-      </Container>
+        <Container sx={{ py: 2, flex: 1}}>
+          <Typography variant="h4" sx = {{textAlign: 'center'}}>Hello</Typography>
+          <Typography variant="body1" sx = {{textAlign: 'center'}}>Explanation stuff</Typography>
+          <Stack alignItems='center'>
+            <Stack direction='row' alignItems={'center'}>
```
Comment (mattrafuse) [inline ui/src/App.tsx]:
```suggestion
            <Stack direction='row' alignItems="center">
```

File: ui/src/App.tsx [hunk 3]
```diff
@@ -64,34 +64,50 @@ function App() {
...
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>A category</Typography>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>Some other category</Typography>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>Looooooooong looooooong multi-line category</Typography>
+            </Stack>
+            <Stack direction='row'>
+              <Stack width='200px' alignItems='center'>
+                <Typography height='200px' variant="body1" sx={{textAlign:'center', alignItems:'center', display:'flex'}}>Author</Typography>
+                <Typography height='200px' variant="body1" sx={{textAlign:'center', alignItems:'center', display:'flex'}}>Some other author with a super-duper multi-line name</Typography>
+                <Typography height='200px' variant="body1" sx={{textAlign:'center', alignItems:'center', display:'flex'}}>One more author</Typography>
```
Comment (mattrafuse) [inline ui/src/App.tsx]:
```suggestion
                <Typography height='200px' variant="body1" sx={{textAlign:'center', alignItems:'center', display:'flex'}}>Author</Typography>
                <Typography height='200px' variant="body1"}>Some other author with a super-duper multi-line name</Typography>
                <Typography height='200px' variant="body1">One more author</Typography>
```

File: ui/src/App.tsx [hunk 4]
```diff
@@ -64,34 +64,50 @@ function App() {
...
+                    '&:nth-child(-n+6)': {
+                      borderBottom: 'thin solid black',
+                    },
+                  },
+                })}
+              >
+                {[...Array(9).keys()].map(entry => <GridButton key={entry}/>)}
+              </Box>
+              {/* Dummy box to take up space: */}
+              <Box height='600px' width='200px'/>
```
Comment (mattrafuse) [inline ui/src/App.tsx]: que?

**Review by mattrafuse — CHANGES_REQUESTED** [review]

File: ui/src/App.tsx (hunk 3 above)
Comment (kerznal) [reply ui/src/App.tsx]: What is the purpose of this suggestion? Won't this remove the centering of the second and third authors' names?

File: ui/src/App.tsx (hunk 4 above)
Comment (kerznal) [reply ui/src/App.tsx]: I wanted the grid itself to be centered along the horizontal axis of the page, so I used the `alignItems='center'` prop for the `Stack` on line 70. However, to the left of the grid is the 200px wide column containing the author names (lines 76-81), so the grid became slightly off-centre. I therefore added this 200px wide "dummy box" on line 107 to the _right_ of the grid, in order to ensure that the grid is in the centre of the page.

I realize it's a bit hack-y. I'd like to have some way to say "centre the grid, and then put the authors column immediately to the left of that." I had hoped to do this using absolute/relative positioning, but I wasn't able to find a solution.

File: ui/src/App.tsx (hunk 1 above)
Comment (kerznal) [reply ui/src/App.tsx]: Why is maxWidth here relevant? My main concern is in ensuring that each component takes up exactly 200px. If I were to use maxWidth, I would have no guarantee that components actually do take up the full 200px that I need them to occupy.

Comment (mattrafuse) [reply ui/src/App.tsx]: Mobile support - 200px on mobile is treated very differently than 200px on desktop.

Comment (mattrafuse) [reply ui/src/App.tsx]: image

**Review by mattrafuse — APPROVED** [review]

**Comment by mattrafuse** [comment]
I've opened a PR off this one if you're interested in taking a look. Take what you like from there!

**Comment by kerznal** [comment]
Closing due to the fact that this was accomplished by #7

**Pull request closed without merging**